mono/metadata/object.c

```diff
@@ -56,6 +56,7 @@
 #include <mono/utils/unlocked.h>
 #include "external-only.h"
 #include "monitor.h"
+typedef guint32 gchandle_t; // FIXME
```

Comment:
This isn't going in.

Reply:
This isn't 
1. the right place for this typedef
2. the right PR to add this typedef

(also global handles are part of the API, so they'll --- at least in part --- need to stay as plain ints, so no destructors).

Reply:
Appears unused in this file/PR. Will confirm/remove. I guess because of the no_safepoints vs. pin. Perhaps that is what you meant.  Agreed.

Reply:
They should have been pointer-sized for implementation flexibilty.
The name-for-documentation-purposes is somewhat useful internally even if stuck with the wrong type externally.
I added it, or expand its scope whenever I add pinning (it is present in some of the code), which was the case here but no longer.
But anyway no longer used here, removed.

Reply:
I guess two names are warranted, maybe `gchandle_raw_t`  w/o destructor and `gchandle_t` with destructor. One can also play game of C and C++ reuse same name but probably bad idea.
(windows.h does this for example with REFIID and nobody seemed to mind except me -- pointer in C, reference in C++; also midl/com does this systematically, but still probably  not worthwhile, better to have the C  subset of  C++ mean the same thing and switching language of  compilation, when in the valid subset, not change meaning, except very subtely and harmlesslyl like our  templates and enum operators).